File: A_PS/pythonPs/P_가장 큰 수.py
def solution(numbers):
    answer = ''
    nums = []
    # permutations는 시간초과 O(n!)
    
    nums = list(map(str, numbers))
    nums.sort(key = lambda x : x*3, reverse=True)
    
    for i in nums:
        answer += i
        
    return str(int(answer))

File: A_PS/javaPs/P_가장큰수.java
package javaPs;

import java.util.*;

public class P_가장큰수 {
    public static void main(String[] args) {
        String[] strArr = new String[]{"2", "6", "10"};
    }
    class Solution {
        public String solution(int[] numbers) {
            String answer = "";

            String[] strArr = new String[numbers.length];

            for(int i=0; i<numbers.length; i++) {
                strArr[i] = String.valueOf(numbers[i]);
            }

            Arrays.sort(strArr, new Comparator<String>() {
                @Override
                public int compare(String o1, String o2) {
                    return (o1 + o2).compareTo(o2 + o1);
                }
            });


            return answer;
        }
    }
}
